provider/server/machine_volume_attach_test.go

```diff
@@ -90,25 +103,301 @@ var _ = Describe("AttachVolume", func() {
+					Driver: "ceph",
+					Handle: listVolumeResp.Volumes[0].Status.Access.Handle,
+					Attributes: map[string]string{
+						"image":    listVolumeResp.Volumes[0].Status.Access.Attributes["image"],
+						"monitors": cephMonitors,
+					},
+					SecretData: map[string][]byte{
+						"userID":  []byte(cephUsername),
+						"userKey": []byte(cephUserkey),
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(attachVolumeResp).NotTo(BeNil())
+
+		By("ensuring volume is attached to the machine domain")
+		var disks []libvirtxml.DomainDisk
+		Eventually(func() int {
+			domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			domainXML := &libvirtxml.Domain{}
+			Expect(domainXML.Unmarshal(domainXMLData)).Should(Succeed())
+			disks = domainXML.Devices.Disks
+			return len(disks)
+		}).WithTimeout(2 * time.Minute).WithPolling(2 * time.Second).Should(Equal(1))
+		Expect(disks[0].Serial).To(HavePrefix("oda"))
+
+		By("ensuring attached volume has been updated in machine status field")
+		Eventually(func() *machineiriv1alpha1.MachineStatus {
+			listResp, err := machineClient.ListMachines(ctx, &machineiriv1alpha1.ListMachinesRequest{
+				Filter: &machineiriv1alpha1.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).NotTo(BeEmpty())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status
+		}).Should(SatisfyAll(
+			HaveField("Volumes", ContainElements(
+				&machineiriv1alpha1.VolumeStatus{
+					Name:   listVolumeResp.Volumes[0].Metadata.Id,
+					Handle: "libvirt-provider.ironcore.dev/ceph/libvirt-provider.ironcore.dev/ceph^" + listVolumeResp.Volumes[0].Status.Access.Handle,
+					State:  machineiriv1alpha1.VolumeState_VOLUME_ATTACHED,
 				})),
-			HaveField("State", Equal(iri.MachineState_MACHINE_RUNNING)),
+			HaveField("State", Equal(machineiriv1alpha1.MachineState_MACHINE_RUNNING)),
 		))
+	})
 
-		By("attaching volume with connection details to a machine")
-		attachVolumeConnectionResp, err := machineClient.AttachVolume(ctx, &iri.AttachVolumeRequest{
+	It("should correctly attach an encrypted volume to the machine", func(ctx SpecContext) {
+		By("creating a machine")
+		createResp, err := machineClient.CreateMachine(ctx, &machineiriv1alpha1.CreateMachineRequest{
+			Machine: &machineiriv1alpha1.Machine{
+				Metadata: &irimetav1alpha1.ObjectMetadata{
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: &machineiriv1alpha1.MachineSpec{
+					Power: machineiriv1alpha1.Power_POWER_ON,
+					Class: machineClassx2medium,
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		DeferCleanup(func(ctx SpecContext) {
+			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &machineiriv1alpha1.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
```

Comment:
unclear using of ctx

Reply:
optional: rename err to locErr or defErr 

i am not big fan of shadowing. Juniors have problem with it and it can very easy change behavior of program when you delete one character ':', so it affects review process because when i see shadowing i have to me careful with error flow